### PR TITLE
Add extension point for impersonate app

### PR DESCRIPTION
**Pull request by sharidas**
This is required as the ajax call made
in the update function was happening too
early and hence the impersonate icons were
not able display on the users page.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

## Description
Adding an extension point for the impersonate app.

## Related Issue

This change is required for https://github.com/owncloud/impersonate/pull/24

## Motivation and Context

This change helps us solve the ajax load happening which causes
the failure to load the impersonate icons.

## How Has This Been Tested?

This change has been tested with the impersonate app.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by sharidas** [comment]
@PVince81 @butonic Kindly review my change.

File: settings/js/users/users.js [hunk 1]
```diff
@@ -690,6 +690,7 @@ var UserList = {
 };
 
 $(document).ready(function () {
+	OC.Plugins.attach('OCA.Impersonate.Impersonate', UserList);
```
Comment (PVince81) [inline settings/js/users/users.js]: Wrong, core **must not** hard-code app names. An extension point is to be extended by any apps, not a specific one.

The extension point must ideally be named after the class it allows to extend. Please rename to "OC.Settings.UserList"

Comment (sharidas) [inline settings/js/users/users.js]: Ok.

**Comment by sharidas** [comment]
Updated the PR with "OC.Settings.UserList"

**Comment by PVince81** [comment]
:+1:

**Pull request merged**